librz/bin/format/mach0/coresymbolication.c

```diff
@@ -117,21 +117,14 @@ RZ_API ut64 rz_coresym_cache_element_pa2va(RzCoreSymCacheElement *element, ut64
 
 static char *str_dup_safe(const ut8 *b, const ut8 *str, const ut8 *end) {
 	if (str >= b && str < end) {
-		int len = rz_str_nlen((const char *)str, end - str);
-		if (len) {
-			return rz_str_ndup((const char *)str, len);
-		}
+		return rz_str_ndup((const char *)str, end - str);
 	}
 	return NULL;
 }
 
-static char *str_dup_safe_fixed(const ut8 *b, const ut8 *str, ut64 len, const ut8 *end) {
+static char *str_ndup_safe(const ut8 *b, const ut8 *str, ut64 len, const ut8 *end) {
```

Comment:
We should have these functions in the RzUtil API, IMHO.

Reply:
not really, the rz_str_ndup is already safe.